query/logicalplan/validate.go

```diff
@@ -0,0 +1,234 @@
+package logicalplan
+
+import (
+	"fmt"
+	"github.com/apache/arrow/go/v8/arrow/scalar"
+	"github.com/segmentio/parquet-go/format"
+	"reflect"
+	"strings"
+)
+
+// PlanValidationError is the error representing a logical plan that is not valid.
+type PlanValidationError struct {
+	message  string
+	plan     *LogicalPlan
+	children []*ExprValidationError
+}
+
+// PlanValidationError.Error prints the error message in a human-readable format.
+// this method allows the type to implement the error interface
+func (e *PlanValidationError) Error() string {
```

Comment:
The way the errors are printed might change a bit. I didn't test this too much, so the current logic is kind of a placeholder. If we do indeed want these errors to print in a human readable format, I can play around with it a bit to try to figure out some format that makes sense.